cnd/src/jsonrpc.rs

```diff
@@ -23,28 +16,32 @@ impl Client {
         }
     }
 
-    pub async fn send<Req, Res>(&self, request: Request<Req>) -> Result<Res, Error>
+    pub async fn send<Req, Res>(&self, request: Request<Req>) -> anyhow::Result<Res>
     where
-        Req: Serialize,
+        Req: Debug + Serialize,
         Res: DeserializeOwned,
     {
+        let url = self.url.clone();
         let response = self
             .inner
-            .post(self.url.clone())
+            .post(url.clone())
             .json(&request)
             .send()
             .await?
             .json::<Response<Res>>()
-            .await?;
+            .await
+            .context("failed to deserialize JSON response as JSON-RPC response")?;
 
         match response {
             Response::Success { result } => Ok(result),
-            Response::Error { code, message } => Err(Error::JsonRpc { code, message }),
+            Response::Error(error) => {
+                Err(error).with_context(|| format!("JSON-RPC request {:?} failed", request))
```

Comment:
Is this `with_context` necessary? I think you are just duplicating the message that is on the `JsonRpcError` struct.

Reply:
My intention is to give the context of which _request_ resulted in a JSON-RPC _response_ error. Request contains method and params, whereas the response error only has code and message.

Reply:
Aha! That makes sense :)